src/transformers/pipelines/__init__.py

```diff
@@ -945,7 +959,7 @@ def pipeline(
     load_tokenizer = load_tokenizer and pipeline_class._load_tokenizer
     load_feature_extractor = load_feature_extractor and pipeline_class._load_feature_extractor
     load_image_processor = load_image_processor and pipeline_class._load_image_processor
-    load_processor = load_processor and pipeline_class._load_processor
+    load_processor = load_processor or pipeline_class._load_processor
```

Comment:
Change this to `or` because unlike `load_tokenizer`, `load_image_processor` and `load_feature_extractor` which can be set to True in the logic right afterwards depending on the type of data involved in the pipeline, if `load_processor` is False at this point, it won't change.
This means that if the model is not in `PROCESSOR_MAPPING`, the processor won't be loaded, and the logic below to         "try to infer processor from model or config name" won't be executed.
This is a problem for models who share an architecture with other models, such as Donut, and thus can't be in PROCESSOR_MAPPING. Changing this to `or` fixes the problem, but I would like to make sure I don't break anything by doing this. @qubvel I think you added this logic so I would be glad to have your input on this

Reply:
Update: I am reverting this change as it is causing some problems in other tests. I'll open a PR for it later on.